About dev section (#64)
madewithkoji/docs: src/asciidoc/docs/about/contribute.adoc

	`:page-banner: \{imagesDir\}/example-page.png`

[NOTE]
====
- `\{imagesDir\}` refers to the image directory, which can be found in the repository at `static/images`
- If the page contains block images, you don't need to set the page banner attribute. The first block image on the page is used, instead.
====

=== Tabs
Sometimes it can be useful to sort your content into tabs.
You'll see them used particularly often to display code snippets for differing frameworks.
To add tabs into your document use the following syntax:

[source, asciidoc]
----
include::../_includes/tabs-example.adoc[tags=*]
----

The above code will output the following:

include::../_includes/tabs-example.adoc[tags=*]

You will also notice that on some pages changing tabs will switch the tabs in multiple places (for example, choosing a coding framework in a blueprint article). 
To sync multiple sets of tabs, use the `scope` attribute. Then, selecting a tab on the page also selects any other tabs with the same scope and a matching tab title.

[source, asciidoc]
----
include::../_includes/tabs-scope-example.adoc[tags=*]
----

include::../_includes/tabs-scope-example.adoc[tags=*]

=== Documentation structure

> Understanding the file structure for the documentation will also be helpful for making modifications.

Reviewer: ```suggestion
Understanding the file structure of the documentation is helpful when making modifications.
```
Just a style thing to avoid future tense ("will") when possible

> Whilst all the documentation is structured as per the site under `src/asciidoc/` there are two other file structures to understand that will help you build content:

Reviewer: Replace "whilst". We're standardizing on US English :)

nav.json::

> 	The navigation of the site is handled in a single json file `src/nav.json`. This structure uses the `name` and `slug` attributes to define where the pages sit in the navigation both for the sidebar and the overall url structure.

Reviewer: ```suggestion
	The site navigation is defined in the `src/nav.json` file. 
	The `name` and `slug` attributes define where a page is listed in the sidebar as well as the URL structure.
```
Made a few edits to short and also apply some standards (URL vs. url)

= Contributing to the Koji developer site
:page-slug: contribute-koji-developers
:page-banner: {imagesDir}/example.png

The Koji for Developers documentation site is built and maintained by developers just like you.
You too can help by contributing to the site *structure*, writing new *articles*, or *suggesting edits* to existing documentation.

Anyone can open an issue about documentation, or contribute a change with a pull request to the `madewithkoji/docs` https://github.com/madewithkoji/docs[GitHub repository].

link to code of conduct?

include::../_includes/tip-discord-server.adoc[tags=*]

== Editing the documentation

> The documentation is written in https://asciidoc.org/[Asciidoc] which is a lightweight markup language for simplifying the creation of documentation.

Reviewer: ```suggestion
The documentation is written in https://asciidoc.org/[Asciidoc], which is a lightweight markup language for simplifying the creation of documentation.
```

All the AsciiDoc files that map to this site can be found in the repository under `src/asciidoc/`.

Here's a quick example:

[source, asciidoc]
----
= Koji for Developers
Create your own templates for https://withkoji.com[Koji].

> 

Reviewer: Suggest adding a space after the h1 and removing the extra space below the preamble (to demonstrate Asciidoc standards in the example).

== Use any framework

* React
* Vanilla JS

[source,javascript]
console.log('Hello World');
----

To see more and get a better understanding of the syntax visit the https://powerman.name/doc/asciidoc[AsciiDoc cheatsheet] and the https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/[AsciiDoc Syntax Quick Reference].

> We also have are a few extensions on AsciiDoc syntax added in to make the developer site ever better.

Reviewer: ```suggestion
We have also extended the AsciiDoc syntax to make the developer site ever better.
```
Think we can streamline a little. Here's one suggestion

> === Page Attributes

Reviewer: ```suggestion
=== Page attributes
```
Sentence case for titles (trying to maintain that as a standard)

Page attributes are used to add extra data related to a document.
Add these to the top of your document, just under the document title.

Page slug::

> 	This corresponds to the page url and is used for linking between different pages. +

Reviewer: ```suggestion
	Corresponds to the page URL and is used for linking between different pages. +
```
Also, re: your question about explaining the structure, we might want to mention `nav.json` and the `_includes` directory (not here, but we might link to the nav.json info from here)

	`:page-slug: example-page`
Page description::

> 	This is a short description of the content of the page and is used when sharing the link on social media and chat platforms. +

Reviewer: ```suggestion
	Provides a short description of the content of the page and is used when sharing the link on social media and chat platforms. +
```
Just making the list structure parallel and removing "this"

	`:page-description: This page is an example`
Page banner::

> 	This references an image which is used as the thumbnail when sharing the link on social media and chat platforms. +

Reviewer: ```suggestion
	References an image, which is used as the thumbnail when sharing the link on social media and chat platforms. +
```

	`:page-banner: \{imagesDir\}/example-page.png`

NOTE: `\{imagesDir\}` refers to the image directory, which can be found in the repository at `static/images`

> TIP: If you have block images on your page you don't need to set the page banner, the first image will be used instead

Reviewer: ```suggestion
TIP: If the page contains block images, you don't need to set the page banner attribute. The first block image on the page is used, instead.
```
Also, wondering if we should combine this tip into the note above (maybe as two bullets in the same note). Just to avoid having back-to-back admonitions.

=== Tabs
Sometimes it can be useful to sort your content into tabs.
You'll see them used particularly often to display code snippets for differing frameworks.
To add tabs into your document use the following syntax:

[source, asciidoc]
----
[.tabs]
--

.Tab 1 title
Tab 1 content.

.Tab 2 title
Tab 2 content.

--
----

The above code will output the following:

[.tabs]
--

.Tab 1 title
Tab 1 content.

.Tab 2 title
Tab 2 content.

--

> You will also notice that on some pages changing tabs will switch the tabs in multiple places (e.g. choosing a coding framework in a blueprint article). You can use scope attribute to sync multiple sets of tabs with one another, by setting two sets of tabs with the same scope attributes, whenever a tab is selected all the tabs with the same name will also be selected across matching scope names.

Reviewer: ```suggestion
You will also notice that on some pages changing tabs will switch the tabs in multiple places (for example, choosing a coding framework in a blueprint article). 
To sync multiple sets of tabs, use the `scope` attribute and matching tab titles. Then, selecting one tab on the page also selects any tabs with the same scope and the same title.
```
This paragraph is a little confusing. Think the suggestion needs some word-smithing, too, but hopefully gives you some ideas

[source, asciidoc]
----
[.tabs,scope="framework"]
--

.React
Some react code

.Vanilla JS
Some Vanilla JS code

--

> [.tabs,scope="framework"]

Reviewer: Let's use "lang" in the example since that's the scope we're using on the site.

--

.React
Some more react code. These tabs will sync with the above ones.

.Vanilla JS
Some more Vanilla JS code. These tabs will sync with the above ones.

--
----

[.tabs,scope="framework"]
--

.React
Some react code

.Vanilla JS
Some Vanilla JS code

--

[.tabs,scope="framework"]
--

.React
Some more react code. These tabs will sync with the above ones.

.Vanilla JS
Some more Vanilla JS code. These tabs will sync with the above ones.

--

== Upgrading the site

You can also contribute by making improvements to how the developer site run. The developer site is built using https://www.gatsbyjs.com/[Gatsby], https://asciidoctor.org/docs/asciidoctor.js/[Asciidoctor.js] and the https://www.gatsbyjs.com/plugins/gatsby-transformer-asciidoc/[gatsby asciidoc transformer].

> Download https://github.com/madewithkoji/docs[the GitHub repository] and start your development server by running `npm start`.

Reviewer: I think we should add some instructions on how to preview the docs/site with Gatsby (gatsby develop and gatsby build / gatsby serve). We could also add a recommended step to the submission process for previewing the changes.
